src/persist-client/src/batch.rs

```diff
@@ -131,17 +135,12 @@ where
     /// marks them as deleted.
     #[instrument(level = "debug", skip_all, fields(shard = %self.shard_id))]
     pub async fn delete(mut self) {
-        // TODO: This is temporarily disabled because nemesis seems to have
-        // caught that we sometimes delete batches that are later needed.
-        // Temporarily removing the deletions while we figure out the bug in
-        // case it has anything to do with CI timeouts.
-        //
-        // for key in self.blob_keys.iter() {
-        //     retry_external("batch::delete", || async {
-        //         self.blob.delete(key).await
-        //     })
-        //     .await;
-        // }
+        for part in self.batch.parts.iter() {
```

Comment:
other thought is wrapping this in a feature flag if we're rolling this out before we're confident S3 versioning won't have to be disabled on prod for any reason